### ISSUE TITLE
Remain `mrb_convert_type` prototype and introduce newly named function in mruby 3

### ISSUE THREAD

**Issue by udzura** [mruby/mruby#5070]
Found a breaking change in mruby 3:

https://github.com/mruby/mruby/commit/3b6b8d3a4bf84f10b1fcb295dfbb4bba2061bc1f

This affects so many mgems (including like mruby-env and  mruby-process... which are referenced in many mruby products)!!

IMO I agree with your intention that current prototype requires a redundant argument such like `"String"`, so how about introducing newly named function like `mrb_convert_type_sym` ?

**Comment by udzura** [comment]
Or, if `version.h` is bumped in mruby3 branch, we can send P/Rs absorbing incompatibility to these mgems in advance:

```c
#if MRUBY_RELEASE_MAJOR >= 3
  #define mrb_convert_type_compat(mrb, val, type, tname, method) \
    mrb_convert_type(mrb, val, type, MRB_SYM(method))
#else
  #define mrb_convert_type_compat(mrb, val, type, tname, method) \
    mrb_convert_type(mrb, val, type, tname, method)
#end

//...
// change usage of mrb_convert_type() below
```

**Comment by matz** [comment]
OK. I will rename `mrb_convert_type()` and `mrb_check_convert_type()` to `mrb_type_convert()` and `mrb_type_convert_check()` respectively, and prepare macros of old names with same arguments.

**Comment by matz** [comment]
Fixed  by ba1ebd3c4bf6

**Comment by udzura** [comment]
Thanks for your quick response!
